lib/rollout.rb

```diff
@@ -1,28 +1,30 @@
 require "rollout/version"
 require "zlib"
 require "set"
+require "json"
 
 class Rollout
   class Feature
-    attr_accessor :groups, :users, :percentage
+    attr_accessor :groups, :users, :percentage, :data
     attr_reader :name, :options
 
     def initialize(name, string = nil, opts = {})
       @options = opts
       @name    = name
 
       if string
-        raw_percentage,raw_users,raw_groups = string.split("|")
+        raw_percentage,raw_users,raw_groups,raw_data = string.split(/(?<!\\)\|/)
         @percentage = raw_percentage.to_f
         @users = (raw_users || "").split(",").map(&:to_s).to_set
         @groups = (raw_groups || "").split(",").map(&:to_sym).to_set
+        @data = raw_data ? JSON.parse(raw_data) : {}
```

Comment:
`String#blank?` is a method from ActiveSupport. Instead of importing that in, I just implemented the same functionality with `raw_data.nil? || raw_data.strip.empty?`